### PR TITLE
Update xorm and fix dump command

### LINKED ISSUE #686: Gitea dump not working
- Gitea version (or commit ref): 1.0.0+129-g6dd096b7
- Git version: n.a.
- Operating system: Official Docker image
- Database:
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:
Running `gitea dump -d postgres` yields:
https://gist.github.com/mjwwit/bbc97d024a12a286f27cd75b11ac5376

## Description
Trying to dump Gitea data using `gitea dump` on a gitea setup using sqlite3 causes the `SIGSEGV` error in the gist linked above. The output database format does not matter, the same error seems to happen regardless of the `-d` flag value (it even happens when the flag is absent).

### PR DESCRIPTION
**Pull request by lunny**
This will resolve #686

### CODE REVIEW

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by bkcsoft** [comment]
@lunny I don't see how this "fixes" #686. It only fails when there's an error, it doesn't fix the error itself?

**Comment by lunny** [comment]
The mainly bug is in XORM. When dump from Sqlite to Postgres, the bool type transform was not corrected.

**Comment by bkcsoft** [comment]
LGTM

File: vendor/vendor.json [hunk 1]
```diff
@@ -243,10 +243,10 @@
 			"revisionTime": "2016-08-11T02:11:45Z"
 		},
 		{
-			"checksumSHA1": "eEVm8NprkjOX2HZVxIJwbOLD5Hg=",
+			"checksumSHA1": "kAy5B68FSqoQNXWGnwdmmzZ0ElQ=",
 			"path": "github.com/go-xorm/xorm",
-			"revision": "44b70cccf417e3d4a565aaa3128853b5cd553767",
-			"revisionTime": "2017-01-03T05:31:47Z"
+			"revision": "263f9d3fa5d9ce24069ffdc9486c58e370f4f707",
```
Comment (Bwko) [inline vendor/vendor.json]: @lunny This should be go-xorm/xorm@c360c20aa58a6dec86084967e93f673c57278da1

Comment (lunny) [reply vendor/vendor.json]: Done.

**Comment by Bwko** [comment]
LGTM

**Comment by mjwwit** [comment]
Tested by migrating a sizable production instance from SQLite3 to Postgres, and it worked like a charm!

**Pull request merged**